webhook/resources/node/mutator.go

```diff
@@ -35,29 +37,55 @@ func (n *nodeMutator) Resource() admission.Resource {
 }
 
 func (n *nodeMutator) Create(request *admission.Request, newObj runtime.Object) (admission.PatchOps, error) {
-	return mutateNode(newObj)
+	newNode := newObj.(*longhorn.Node)
+
+	return mutateNode(newNode)
 }
 
 func (n *nodeMutator) Update(request *admission.Request, oldObj runtime.Object, newObj runtime.Object) (admission.PatchOps, error) {
-	return mutateNode(newObj)
+	newNode := newObj.(*longhorn.Node)
+
+	return mutateNode(newNode)
 }
 
-func mutateNode(newObj runtime.Object) (admission.PatchOps, error) {
+func mutateNode(node *longhorn.Node) (admission.PatchOps, error) {
 	var patchOps admission.PatchOps
 
-	node := newObj.(*longhorn.Node)
-
 	if node.Spec.Tags == nil {
 		patchOps = append(patchOps, `{"op": "replace", "path": "/spec/tags", "value": []}`)
+	} else {
+		if len(node.Spec.Tags) > 0 {
+			tags, err := util.ValidateTags(node.Spec.Tags)
+			if err != nil {
+				return nil, err
+			}
+			bytes, err := json.Marshal(tags)
+			if err != nil {
+				return nil, err
+			}
+			patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/spec/tags", "value": %s}`, string(bytes)))
+		}
```

Comment:
Should Longhorn validate the tags in the validator rather than the mutator? Then is the following `json.Marshal()` required?

Similar below.

Reply:
is the main mutation in the function `sort.Strings(tags)`? If YES, I think we can let this mutator take over the sorting only.

Reply:
Yes. Sounds good. Will follow your suggestion.